.dev/edge-native-base/src/oauth2_proxy/app.py

```diff
@@ -22,14 +24,21 @@
 LOG.addHandler(handler)
 LOG.setLevel(logging.DEBUG)
 
+
 # When run from Edge, these environment variables will be provided
 API_TOKEN = os.environ.get("JUPYTERHUB_API_TOKEN")
 API_URL = os.environ.get("JUPYTERHUB_API_URL")
+ACTIVITY_URL = os.environ.get("JUPYTERHUB_ACTIVITY_URL")
 JUPYTERHUB_SERVICE_PREFIX = os.environ.get("JUPYTERHUB_SERVICE_PREFIX")
-SESSION_SECRET_KEY = os.environ.get("SESSION_SECRET_KEY", "super secret key")
-EDGE_API_SERVICE_URL = os.environ.get("EDGE_API_SERVICE_URL", None)
-EDGE_API_ORG = os.environ.get("EDGE_API_ORG", None)
-NATIVE_APP_MODE = os.environ.get("NATIVE_APP_MODE", None)
+SERVER_NAME = os.environ.get("JUPYTERHUB_SERVER_NAME", "")
+HAVE_JUPYTERHUB = API_TOKEN is not None
```

Comment:
Suggest

```
HAVE_JUPYTERHUB = API_TOKEN is not None and API_URL is not None
```

Reply:
When would they not both be valid/both be None?

Reply:
I think this is a question of guarantees from JupyterHub. It is supposed to provide all three from JupyterHub, but JUPYTERHUB_API_TOKEN is most likely to be erroneously provided in container mode by a user who is straying from the guardrails. My worry is if a user provides this token because they are conflating EdgeSession documentation with getting this container to function.